src/components/Category.js

```diff
@@ -0,0 +1,37 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+
+const Category = ({ registers, name }) => {
+  const [showButton, setShowButton] = useState(false);
+
+  const RegisterCategory = ({ regName }) => (
+    <button type="submit" onClick={() => setShowButton(!showButton)}>
+      {regName}
+    </button>
+  );
+
+  RegisterCategory.propTypes = {
+    regName: PropTypes.string.isRequired,
+  };
+
+  if (!showButton) {
+    return (
+      <div>
+        <RegisterCategory regName={name} />
+      </div>
+    );
+  }
+  return (
+    <div>
+      <RegisterCategory regName={name} />
+      {registers}
+    </div>
+  );
+};
+
+Category.propTypes = {
+  registers: PropTypes.arrayOf(PropTypes.object).isRequired,
```

Comment:
instead of just defining `PropTypes.object`, could we explicitly set the shape of the object? Look into `PropTypes.shape` or see https://reactjs.org/docs/typechecking-with-proptypes.html

Reply:
will look into this

Reply:
i set the shape of objects inside the registers array here.
inside Category.js, registers contains an array of complicated li elements, so I left object as the type there for the sake of simplicity